server/sonar-server/src/main/java/org/sonar/server/authentication/JwtHttpHandler.java

```diff
@@ -0,0 +1,174 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.authentication;
+
+import static java.util.Objects.requireNonNull;
+import static org.elasticsearch.common.Strings.isNullOrEmpty;
+import static org.sonar.server.authentication.CookieUtils.findCookie;
+
+import com.google.common.collect.ImmutableMap;
+import io.jsonwebtoken.Claims;
+import java.util.Date;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang.time.DateUtils;
+import org.sonar.api.config.Settings;
+import org.sonar.api.platform.Server;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.System2;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.user.UserDto;
+
+@ServerSide
+public class JwtHttpHandler {
+
+  private static final String SESSION_TIMEOUT_PROPERTY = "sonar.authentication.timeoutInHours";
+  private static final int SESSION_TIMEOUT_DEFAULT_VALUE = 72;
+
+  private static final String JWT_COOKIE = "JWT-SESSION";
+  private static final String LAST_REFRESH_TIME_PARAM = "lastRefreshTime";
+
+  private static final String CSRF_JWT_PROPERTY = "xsrfToken";
+
+  // Time after which a user will be disconnected
+  private static final int SESSION_DISCONNECT_IN_SECONDS = 3 * 30 * 24 * 60 * 60;
+
+  // This refresh time is used to refresh the session
+  // The value must be lower than SESSION_TIMEOUT_IN_SECONDS
+  private static final int SESSION_REFRESH_IN_SECONDS = 5 * 60;
+
+  private static final String RAILS_USER_ID_SESSION = "user_id";
+
+  private final System2 system2;
+  private final DbClient dbClient;
+  private final Server server;
+  private final JwtSerializer jwtSerializer;
+
+  // This timeout is used to disconnect the user we he has not browse any page for a while
+  private final int sessionTimeoutInSeconds;
+  private final JwtCsrfVerifier jwtCsrfVerifier;
+
+  public JwtHttpHandler(System2 system2, DbClient dbClient, Server server, Settings settings, JwtSerializer jwtSerializer, JwtCsrfVerifier jwtCsrfVerifier) {
+    this.jwtSerializer = jwtSerializer;
+    this.server = server;
+    this.dbClient = dbClient;
+    this.system2 = system2;
+    this.sessionTimeoutInSeconds = getSessionTimeoutInHours(settings) * 60 * 60;
```

Comment:
`getSessionTimeoutInHours()` is called only here, so why converting result ? Method should be refactored to `getSessionTimeoutInSeconds()`.